books/RayTracingTheRestOfYourLife.html

```diff
@@ -423,184 +571,504 @@
+distributions is used somewhat commonly in practice, although there are much more efficient means
+of doing so than a simple binary partition. If you have an arbitrary function that you wish to use
+as the PDF for a distribution, you'll want to research the _Metropolis-Hastings Algorithm_.
+
+Approximating Distributions
+---------------------------
+
+This was a lot of math and work to build up a couple of notions. Let's return to our initial PDF and
+enumerate a new term. Where we don't explicitly set a probability, we assume to be zero. So for our
+example from the beginning of the chapter, $p(r) = 0$, for $r < 0$ and $r > 2$. _i.e._, the
+probability of generating a random number there is zero. We can rewrite our $p(r)$ in piecewise
+fashion:
+
+    $$ p(r)=\begin{cases}0 & r < 0 \\ \frac{r}{2} & 0 < r < 2 \\ 0 & r > 2  \\  \end{cases} $$
+
+If you consider what we were trying to do in the previous section, a lot of math revolved around the
+_accumulated_ area (or _accumulated_ probability) from zero. In the case of the function
+
+    $$ f(x) = e^{\frac{-x}{2 \pi}} sin^2(x)  $$
+
+we cared about the accumulated probability from $0$ to $2\pi$ (100%) and the accumulated probability
+from $0$ to $2.017$ (50%). We can generalize this to an important term, the
+_Cumulative Distribution Function_ $P(x)$ is defined as:
+
+    $$ P(x) =  \int_{-\infty}^{x}  p(x') dx' $$
+
+Or,
+
+    $$ P(x) = \text{area}(p(x'), -\infty, x) $$
+
+Which is the amount of _cumulative_ probability from $-\infty$. If we take the integration outlined
+above, we get the piecewise $P(r)$:
+
+    $$ P(r)=\begin{cases}0 & r < 0 \\ \frac{r^2}{4} & 0 < r < 2 \\ 1 & r > 2  \\  \end{cases} $$
+
+The _Probability Density Function_ (PDF) is the probability function that explains how likely an
+interval of numbers is to be chosen. The _Cumulative Distribution Function_ (CDF) is the
+distribution function that explains how likely all numbers smaller than its input is to be chosen.
+To go from the PDF to the CDF, you need to integrate from $-\infty$ to $x$, but to go from the CDF
+to the PDF, all you need to do is take the derivative:
+
+    $$ p(x) = \frac{d}{dx}P(x) $$
 </div>
 
 <div class='together'>
-This says _the probability that a random variable with our PDF is less than one is 25%_. This gives
-rise to a clever observation that underlies many methods to generate non-uniform random numbers. We
-want a function `f()` that when we call it as `f(random_double())` we get a return value with a PDF
-$\frac{x^2}{4}$. We don’t know what that is, but we do know that 25% of what it returns should be
-less than 1.0, and 75% should be above 1.0. If $f()$ is increasing, then we would expect $f(0.25) =
-1.0$. This can be generalized to figure out $f()$ for every possible input:
+If we evaluate the CDF, $P(r)$, at $r = 1.0$, we get:
 
-  $$ f(P(x)) = x $$
+    $$ P(1.0) = \frac{1}{4} $$
+
+This says _a random variable plucked from our PDF has a 25% change of being 1 or lower_. We want a
```

Comment:
chance

Reply:
Which reminds me of a dad joke I just ran across. I wanted to become a Gregorian monk, but I never got the chants.

Reply:
At this rate, none of us will get the chants.